src/jit/gentree.h

```diff
@@ -4037,26 +3884,198 @@ struct GenTreeIndir : public GenTreeOp
 #endif
 };
 
-// gtObj  -- 'object' (GT_OBJ). */
+// gtBlk  -- 'block' (GT_BLK, GT_STORE_BLK).
+//
+// This is the base type for all of the nodes that represent block or struct
+// values.
+// Since it can be a store, it includes gtBlkOpKind to specify the type of
+// code generation that will be used for the block operation.
+
+struct GenTreeBlk : public GenTreeIndir
+{
+public:
+    // The data to be stored (null for GT_BLK)
+    GenTree*& Data()
+    {
+        return gtOp2;
+    }
+    void SetData(GenTree* dataNode)
+    {
+        gtOp2 = dataNode;
+    }
```

Comment:
No, it is not always null. These get transformed to `GT_STORE_BLK`, etc. by `Rationalizer`. Note that above it clearly indicates that this node is used for `GT_STORE_BLK` as well (and, by inheritance, by `GT_STORE_OBJ` and `GT_STORE_DYN_BLK`).

---
In reply to: [77278869](https://github.com/dotnet/coreclr/pull/7027#discussion_r77278869) [](ancestors = 77278869)